terraform/files/bin/create_cluster.sh

```diff
@@ -103,16 +107,24 @@ clusterctl $KCCCFG generate cluster "${CLUSTER_NAME}" --list-variables --from ${
 echo "# rendering clusterconfig from template"
 unset CLUSTER_EXISTS
 if test -e ~/${CLUSTER_NAME}/${CLUSTER_NAME}-config.yaml; then
-	echo " Overwriting config for ${CLUSTER_NAME}"
-	CLUSTERS=$(kubectl get clusters | grep -v '^NAME' | grep "^$CLUSTER_NAME " | awk '{ print $1; }')
-	if test -n "$CLUSTERS"; then
-		export CLUSTER_EXISTS=1
-		echo -e " Warning: Cluster exists\n Hit ^C to interrupt"
-		sleep 6
-	fi
+  echo " Overwriting config for ${CLUSTER_NAME}"
+  CLUSTERS=$(kubectl get clusters --all-namespaces | grep -v '^NAME' | grep "$CLUSTER_NAME " | awk '{ print $2; }')
+  if test -n "$CLUSTERS"; then
+    export CLUSTER_EXISTS=1
+    echo -e " Warning: Cluster exists\n Hit ^C to interrupt"
+    sleep 6
+    # If the cluster already exists check if it is in a separate namespace if so we change to default namespace
+    if test -n "$CLUSTER_EXISTS"; then
+      CLUSTER_NAMESPACE=$(kubectl get clusters --all-namespaces | grep -v '^NAME' | grep "$CLUSTER_NAME " | awk '{ print $1; }')
+      if test -n "$CLUSTER_NAMESPACE" -a "$CLUSTER_NAMESPACE" != "default"; then
```

Comment:
I tried it with cluster deployed first in the default namespace and rerun with *create_cluster.sh* it doesn't work for me, shouldn't be an equal sign here `-a "$CLUSTER_NAMESPACE" != "default"`?

Reply:
Reworked completly